src/main/java/hudson/plugins/ec2/util/KeyHelper.java

```diff
@@ -152,4 +154,28 @@ public static String getSshAlgorithm(@NonNull PublicKey serverKey) {
             Properties.removeThreadOverride(Properties.EMULATE_ORACLE);
         }
     }
+
+    /**
+     * Computes the MD5 fingerprint of the given server public key.
+     * The fingerprint is formatted as a colon-separated hexadecimal string.
+     *
+     * @param serverKey The server's {@link PublicKey} object for which the fingerprint needs to be computed.
+     * @return A {@code String} representing the MD5 fingerprint of the given server key in
+     *         colon-separated hexadecimal format, or an empty string if an error occurs during computation.
+     */
+    public static String getFingerprint(@NonNull PublicKey serverKey) {
+        // Emulate Oracle so that the algorithm returned by
+        // org.bouncycastle.jcajce.provider.asymmetric.edec.BCEdDSAPublicKey.getAlgorithm
+        // is the one expected by org.apache.sshd.common.config.keys.KeyUtils
+        try {
+            Properties.setThreadOverride(Properties.EMULATE_ORACLE, true);
+            // Generate MD5 fingerprint just like ssh-keygen
+            byte[] rawFingerprint = KeyUtils.getRawFingerprint(BuiltinDigests.md5.get(), serverKey);
+            return BufferUtils.toHex(':', rawFingerprint).toLowerCase();
+        } catch (Exception e) {
+            return "";
```

Comment:
Is some exception expected here? Seems like it should at least log a warning.

Reply:
Not expected. It all goes does to implementation of Digest / decoder I guess.
We can add a warning.